ql/instruments/basketoption.hpp

```diff
@@ -36,7 +36,7 @@ namespace QuantLib {
       private:
         boost::shared_ptr<Payoff> basePayoff_;
       public:
-        BasketPayoff(const boost::shared_ptr<Payoff> &p)
+        explicit BasketPayoff(const boost::shared_ptr<Payoff> &p)
```

Comment:
Strictly speaking some of the other changes will also break backward compatibility. IHMO code should not depend on single argument constructors for conversion unless there is a very good reason (like e.g. Disposable). But this is a more general code style question, shall we fix the single argument constructors? What do you think?